app/src/routes/project/LayersPanel/Tree/TreeNode.tsx

```diff
@@ -146,21 +156,36 @@ const TreeNode = observer(
                 ) : (
                     <NodeIcon iconClass="w-3 h-3 ml-1 mr-2" node={node.data} />
                 )}
-                <span
-                    className={clsx(
-                        'truncate w-full',
-                        instance
-                            ? selected
-                                ? 'text-purple-100'
-                                : hovered
-                                  ? 'text-purple-200'
-                                  : 'text-purple-300'
-                            : '',
+                <Tooltip>
+                    <TooltipTrigger asChild>
+                        <span
+                            ref={textRef}
+                            className={clsx(
+                                'truncate w-full',
+                                instance
+                                    ? selected
+                                        ? 'text-purple-100'
+                                        : hovered
+                                          ? 'text-purple-200'
+                                          : 'text-purple-300'
+                                    : '',
+                            )}
+                        >
+                            {instance?.component
+                                ? instance.component
+                                : node.data.tagName.toLowerCase()}{' '}
+                            {node.data.textContent}
+                        </span>
+                    </TooltipTrigger>
+                    {isOverflowing && (
```

Comment:
I would just show the tooltip if there's text content for simplicity
https://github.com/onlook-dev/onlook/pull/298/commits/09ce88b6e7cc9ecf0f8c24b1dfaa4ad5190777f2#diff-eced7d84f55e60e45f2876a4fc40b812bb12b845d5b5cdb4d641a619346d96d1R181

Reply:
I think we'll need it for every type of layer

<img width="309" alt="image" src="https://github.com/user-attachments/assets/34c26fa9-ba3d-4394-afcd-ee4599a79236">